### PR TITLE
[PM-5148] Removing Browser Autofill Override Setting and Add Message

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

In reviewing the limitations of the chrome.privacy API, we found that facilitating an extension-based feature to override the default browser settings on Chrome was inadequate. In effect, we can disable certain parts of the autofill features within Chrome, but cannot disable others (such as UI elements tied to the recently integration Password Manager feature). 

As a result, we’ve decided to backpedal on incorporating the Override browser auto-fill settings feature within the browser extension. Instead, we will be placing a message and link below the Show auto-fill menu on form fields setting that encourages users to disable their browser’s auto-fill settings manually.

## Code changes

- **apps/browser/src/_locales/en/messages.json:** Adding text content to present to the user under the `Show auto-fill menu on form fields` setting to guide them towards disabling their browser password manager manually.
- **apps/browser/src/autofill/popup/settings/autofill.component.html:** Removing the `Override browser auto-fill settings` option and adding the message to the auto-fill overlay setting.
- **apps/browser/src/autofill/popup/settings/autofill.component.ts:** Removing the logic required to trigger the `Override browser auto-fill settings` behavior and modal
- **apps/browser/src/manifest.json:** Removing the optional permission `privacy` which was added to trigger our extension based override of the default auto-fill settings for a browser
- **apps/browser/src/manifest.v3.json:** Removing the optional permission `privacy` which was added to trigger our extension based override of the default auto-fill settings for a browser
- **apps/browser/src/platform/browser/browser-api.ts:** Removing the API calls that would facilitate overriding the default auto-fill behavior for a browser.

## Demo of the changes

https://github.com/bitwarden/clients/assets/16629865/55d27182-b8e9-4686-95b0-37fb6bfb29fc

### CODE REVIEW

**Review by MGibson1 — DISMISSED** [review]
Always nice to delete things. This one is bitter sweet, though. Let's look at the web extensions group to see if there are proposals to add built-in password manager disabling to the api

File: apps/browser/src/_locales/en/messages.json [hunk 1]
```diff
@@ -1027,6 +1027,12 @@
   "showAutoFillMenuOnFormFieldsDescAlt": {
     "message": "Applies to all logged in accounts."
   },
+  "turnOffBrowserBuiltInPasswordManagerSettingsLink": {
+    "message": "Turn off your browser’s built in password manager settings"
+  },
+  "turnOffBrowserBuiltInPasswordManagerSettingsToAvoidConflicts": {
+    "message": "to avoid conflicts."
+  },
```
Comment (jprusik) [inline apps/browser/src/_locales/en/messages.json:1035]: This concatenation is likely to translate improperly between languages

**Review by jprusik — DISMISSED** [review]
I'd like to revisit the [translation concatenation](https://github.com/bitwarden/clients/pull/7110/files#r1416362552), but that task doesn't need to be blocking.

Everything else LGTM 👍

File: apps/browser/src/_locales/en/messages.json (hunk 1 above)
Comment (cagonzalezcs) [reply apps/browser/src/_locales/en/messages.json:1035]: Let me ask @danielleflinn if we could consider setting the text to be present as the link entirely to address this.

**Review by jprusik — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**